### PR TITLE
[WIP] Heatmap example

### PR DESCRIPTION
**Pull request by jmkiley**
Ported [this example](https://www.mapbox.com/mapbox-gl-js/example/heatmap/) from GL-JS. I switched the data source to use the [USGS earthquake live data](https://earthquake.usgs.gov/earthquakes/feed/v1.0/geojson.php).

- [X] Write example
- [x] Add comments
- [ ] Incorporate revisions

screenshot 2017-05-12 11 30 13

### CODE REVIEW

File: Examples/Swift/PointHeatmapExample.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        mapView.delegate = self
+        
+        view.addSubview(mapView)
+    }
+    
+    func mapView(_ mapView: MGLMapView, didFinishLoading style: MGLStyle) {
+        
+        // Parse GeoJSON data from USGS on earthquakes in the past week.
+        DispatchQueue.global().async {
+            guard let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") else { return }
```
Comment (friedbunny) [inline Examples/Swift/PointHeatmapExample.swift]: This URL should never fail to initialize and can be safely force-unwrapped.

File: Examples/Swift/PointHeatmapExample.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            DispatchQueue.main.async {
+                self.displayEarthquakes(url: url, style: style)
+            }
```
Comment (friedbunny) [inline Examples/Swift/PointHeatmapExample.swift]: Can you say more about what the intention is here, re: threading? It appears that you’re creating the URL on a background thread, then doing the actual fetching and rendering on the main thread.

File: Examples/Swift/PointHeatmapExample.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    func displayEarthquakes(url: URL, style: MGLStyle) {
+        let symbolSource = MGLSource(identifier: "symbol-source")
+        let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+        
+        // Set the MGLShapeSourceOptions to allow clustering.
+        let options = [MGLShapeSourceOption.clustered: true,
+                       MGLShapeSourceOption.clusterRadius: 20,
+                       MGLShapeSourceOption.maximumZoomLevel: 15] as [MGLShapeSourceOption : Any]
+        
+        let earthquakeSource = MGLShapeSource(identifier: "earthquakes", url: url, options: options)
```
Comment (friedbunny) [inline Examples/Swift/PointHeatmapExample.swift]: If this is where the remote fetching happens, should there be some protection against it failing?

File: Examples/Swift/PointHeatmapExample.swift (hunk 1 above)
Comment (friedbunny) [inline Examples/Swift/PointHeatmapExample.swift]: The URLs themselves are also different — the objc example uses the static GL-JS version of this feed.

**Comment by jmkiley** [comment]
screenshot 2017-05-23 10 20 09

Changed the opacity and colors, switched to dark style.

File: Examples/Swift/PointHeatmapExample.swift (hunk 3 above)
Comment (jmkiley) [inline Examples/Swift/PointHeatmapExample.swift]: I changed the source to a Mapbox hosted one. Will look into using this live data source for a tutorial/different example.

**Comment by jmkiley** [comment]
Is there anything else to address with this? @friedbunny @captainbarbosa

**Comment by captainbarbosa** [comment]
No, I don't think so @jmkiley - looks good! 👍

**Pull request merged**